storage/innobase/handler/handler0alter.cc

```diff
@@ -8955,9 +8955,14 @@ innobase_rename_or_enlarge_column_try(
 
 #ifdef UNIV_DEBUG
 	switch (mtype) {
+	case DATA_MYSQL:
+		if (!(prtype & DATA_BINARY_TYPE) || user_table->not_redundant()
+		    || col->mbminlen != col->mbmaxlen) {
+			break;
+		}
+		/* fall through */
 	case DATA_FIXBINARY:
 	case DATA_CHAR:
-	case DATA_MYSQL:
 		/* NOTE: we could allow this when !(prtype & DATA_BINARY_TYPE)
 		and ROW_FORMAT is not REDUNDANT and mbminlen<mbmaxlen.
 		That is, we treat a UTF-8 CHAR(n) column somewhat like
```

Comment:
Move the comment inside the `if` that you added above. Ensure that there is `ut_ad(col->mbminlen <= col->mbmaxlen);` somewhere along this code path.
I am not suggesting to change the `!=` in the `if` condition, because I believe that equality comparison could sometimes be compiled to faster code than an ordering comparison.

Reply:
Probably we don't care much about this comparison speed in a debug mode :)

Please check that I did what you meant here.

Reply:
Thanks, it looks better now. I’d still suggest moving the assertion
```c++
ut_ad(col->mbminlen <= col->mbmaxlen);
```
before the `switch` statement, so that we won’t have to duplicate it. This is assuming that for non-string columns, these fields would be 0.